src/playermanager.cpp

```diff
@@ -67,6 +67,39 @@ bool ZEPlayer::IsAdminFlagSet(uint64 iFlag)
 	return !iFlag || (m_iAdminFlags & iFlag);
 }
 
+// CONVAR_TODO
+float flFloodInterval = 0.75; // Amount of time allowed between chat messages acquiring flood tokens
+int iMaxFloodTokens = 3; // Maximum number of flood tokens allowed before chat messages are blocked
+
+bool ZEPlayer::IsFlooding()
+{
+	if (m_bGagged) return false;
+
+	float time = gpGlobals->curtime;
+	float newTime = time + flFloodInterval;
+
+	if (m_flLastTalkTime >= time)
+	{
+		if (m_iFloodTokens >= iMaxFloodTokens)
+		{
+			m_flLastTalkTime = newTime + 3.0;
```

Comment:
Why +3.0 here? Also this should also be a constant/cvar.

Reply:
well SM did it so i was just copying